src/components/structures/TimelinePanel.tsx

```diff
@@ -73,6 +73,12 @@ const debuglog = (...args: any[]): void => {
     }
 };
 
+const overlaysBefore = (overlayEvent: MatrixEvent, mainEvent: MatrixEvent): boolean =>
+    overlayEvent.localTimestamp < mainEvent.localTimestamp;
+
+const overlaysAfter = (overlayEvent: MatrixEvent, mainEvent: MatrixEvent): boolean =>
+    overlayEvent.localTimestamp >= mainEvent.localTimestamp;
```

Comment:
This approach isn't really right for aligning timelines - https://github.com/matrix-org/matrix-js-sdk/issues/3325

Reply:
I think this warrants a spec discussion on how this is meant to be done

Reply:
I've raised this as a problem needing further consideration in the spec repo: https://github.com/matrix-org/matrix-spec/issues/1520

For now though, I would note that the `localTimestamp` approach to aligning overlay events is nothing new introduced by this PR, it's only been refactored here. My intent in extracting the comparison into these two functions, was so that we could easily switch to a better approach if one eventually presents itself. There's also some internal context to this which I'll send you in a DM